rcsbapi/data/schema.py

```diff
@@ -443,6 +454,31 @@ def get_descendant_fields(self, schema_graph, node, visited=None):
             return result[0]
         return result
 
+    def extract_name_description(self, schema_part, parent_name=""):
+        if isinstance(schema_part, dict):
+            if 'name' in schema_part and 'description' in schema_part:
+                name = schema_part['name']
+                description = schema_part['description']
+                if name in self.seen_names:
+                    if parent_name:
+                        name = f"{parent_name}.{name}"
+                else:
+                    self.seen_names.add(name)
+                self.name_description_dict[name] = description
+            for key, value in schema_part.items():
+                new_parent_name = schema_part['name'] if key == 'fields' else parent_name
+                self.extract_name_description(value, new_parent_name)
+        elif isinstance(schema_part, list):
+            for item in schema_part:
+                self.extract_name_description(item, parent_name)
+
+    def find_field_names(self, search_string):
```

Comment:
looks good! I would just add some error handling for if someone puts in a list in because currently it just says no fields match.

You could do something like this:
![image](https://github.com/user-attachments/assets/b4de33e0-ecd3-467c-9fa7-76e61cce1a4c)